mimic/catalog.py

```diff
@@ -33,22 +33,40 @@ def __init__(self, tenant_id, region, endpoint_id, prefix=None,
         self.prefix = prefix
         self.external = external
         self.complete_url = complete_url
+        if internal_url is None:
+            self.internal_url = complete_url
+        else:
+            self.internal_url = internal_url
 
         # Gate Check complete_url
         if self.external and self.complete_url is None:
             raise ValueError(
                 'complete_url must be specified when API is external'
             )
 
-    def url_with_prefix(self, uri_prefix):
+    def url_with_prefix(self, uri_prefix, internal_url=False):
         """
         Generate a URL to this endpoint, given the URI prefix for the service.
 
+        :param text_type uri_prefix: prefix to start the URL with, e.g the
+            Request's Base URL
+        :param boolean internal_url: whether or not to provide the Internal or
+            Public URL. If True, provide the internal URL.
+
+        Note: internal_url is only honored by External APIs, e.g `external` is
```

Comment:
markup nitpick: [there's a `..note:` directive for this sort of thing in Sphinx](http://www.sphinx-doc.org/en/stable/markup/para.html?highlight=note#directive-note)